ui-test/allTestsSuite.ts

```diff
@@ -0,0 +1,74 @@
+import { VSBrowser, WebDriver, SideBarView, ActivityBar, EditorView, By, WebView, TextEditor, Workbench, NotificationType } from "vscode-extension-tester";
+import { expect } from 'chai';
+import { addFolderToWorkspace, removeFolderFromWorkspace } from "./common/commonUtils";
+let path = require('path');
+import { pypiWithVulnsUITest } from './manifestFileTests/pypiWithVulns';
+import { pypiWithoutVulnsUITest } from './manifestFileTests/pypiWithoutVulns';
+import { npmWithVulnsUITest } from './manifestFileTests/npmWithVulns';
+import { npmWithoutVulnsUITest } from "./manifestFileTests/npmWithoutVulns";
+var fs = require('fs');
+let os = require('os');
+import debug from 'debug'
+const log = debug('server');
+import { delay } from "./common/helperUtils";
+import { goWithoutVulnsUITest } from "./manifestFileTests/goWithoutVulns";
+import { mavenWithoutVulnsUITest } from "./manifestFileTests/mavenWithoutVulns";
+import { goWithVulnsUITest } from "./manifestFileTests/goWithVulns";
+import { mavenWithVulnsUITest } from "./manifestFileTests/mavenWithVulns";
+import { checkPIEBTN, checkStatusBarBTN } from "./common/negativeTests";
+import { testWithoutVulns, testWithVulns } from "./common/mainTestUtils";
+var assert = require('assert');
+
+describe('Fabric8-Analytics-VSCode-Extension UI tests', function () {
+    let driver: WebDriver;
+    let homedir: string;
+    let dir = path.resolve("./manifests");
+    before(async function () {
+        driver = VSBrowser.instance.driver;
+        homedir = dir
+        const control = await new ActivityBar().getViewControl('Explorer');
+        const explorerView = await control.openView();
+        addFolderToWorkspace(homedir);
+    });
+
+    describe('checking workspace section', function () {
+        it('workspace sections should not be null for manifests', async function () {
+            this.timeout(5000);
```

Comment:
this.timeout  is just a timeout for the test. 
please explain 'better option'.
but other way is also does the same thing eg. 
it('close all tabs in Editor', async function () {
       ...
    }).timeout(1000);

Reply:
timeout is soft wait other way could be hard wait. something like this: 
```await new Promise(cb => setTimeout(cb, timeInMillisec));```

Reply:
timeout is not waiting for response. It's here to make sure that test should run within that particular time else it would fail.
If I don't add timeout it won't make much of a difference I guess but the tests won't get fail because of more time taken to execute them.